packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -673,17 +672,17 @@ class MultiFrameImageStreamCompleter extends ImageStreamCompleter {
     }
   }
 
-  ui.Codec _codec;
+  ui.Codec? _codec;
```

Comment:
This should not be nullalble - the ctor asserts it's not null. We should remove the != null from hasListeners

Reply:
`_codec` is not directly initialized in the constructor (it is in a `future.then()` so some code could has access before the callback is executed).
I can make the field `late` but we could face LazyInitializationError if `addListener` is call before the async initialisation is done.

WDYT?

Reply:
Ohh I see.  Yes that makes sense, this should just be nullable